### PR TITLE
Implement new SAI Actions

### PR DESCRIPTION
**Pull request by Kitzunu**
Quick implementation of the new SAI Actions from https://github.com/azerothcore/azerothcore-wotlk/commit/8aa80eef181af4babea7dd65b938b5d88de9cd03 since it got pushed today.

It is just a basic implementation and can be improved at another time, but I think it is important that we update the wiki :)

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
@P-kito can you have a look here please?

File: docs/smart_scripts.md [hunk 1]
```diff
@@ -3688,6 +3687,75 @@ This is the probability of the event to occur as a percentage from 0-100. So, if
...
+<td><p>Distance between rings (yards)</p></td>
+<td><p>Distance between each summons in a row (yards)</p></td>
+<td><p>Length of the cone</p></td>
+<td><p>Width of the cone (angle)</p></td>
+<td><p>Legion, BFA, Mop like Custom Cone-Spells</p></td>
+</tr>
+<tr class="odd">
+<td><p>SMART_ACTION_PLAYER_TALK</p></td>
+<td><p>220</p></td>
+<td><p>creature_text.groupid</p></td>
```
Comment (P-Kito) [inline docs/smart_scripts.md:3718]: this is acore_string id instead

File: docs/smart_scripts.md [hunk 2]
```diff
@@ -3688,6 +3687,75 @@ This is the probability of the event to occur as a percentage from 0-100. So, if
...
+<td><p>SMART_ACTION_CONE_SUMMON</p></td>
+<td><p>219</p></td>
+<td><p>entry</p></td>
+<td><p>Duration (ms)<br />
+0 = permanent</p></td>
+<td><p>Distance between rings (yards)</p></td>
+<td><p>Distance between each summons in a row (yards)</p></td>
+<td><p>Length of the cone</p></td>
+<td><p>Width of the cone (angle)</p></td>
+<td><p>Legion, BFA, Mop like Custom Cone-Spells</p></td>
```
Comment (P-Kito) [inline docs/smart_scripts.md:3713]: > Allows you to spawn creatures in a cone (As seen in later expansions). Useful for custom cone aoes

File: docs/smart_scripts.md [hunk 3]
```diff
@@ -3688,6 +3687,75 @@ This is the probability of the event to occur as a percentage from 0-100. So, if
...
+</tr>
+<tr class="even">
+<td><p>SMART_ACTION_CONE_SUMMON</p></td>
+<td><p>219</p></td>
+<td><p>entry</p></td>
+<td><p>Duration (ms)<br />
+0 = permanent</p></td>
+<td><p>Distance between rings (yards)</p></td>
+<td><p>Distance between each summons in a row (yards)</p></td>
+<td><p>Length of the cone</p></td>
```
Comment (P-Kito) [inline docs/smart_scripts.md:3711]: in yards

File: docs/smart_scripts.md [hunk 4]
```diff
@@ -3688,6 +3687,75 @@ This is the probability of the event to occur as a percentage from 0-100. So, if
...
+<td><p>Width of the cone (angle)</p></td>
```
Comment (P-Kito) [inline docs/smart_scripts.md:3712]: (angle: 1°-360°)

File: docs/smart_scripts.md [hunk 5]
```diff
@@ -3688,6 +3687,75 @@ This is the probability of the event to occur as a percentage from 0-100. So, if
...
+</p></td>
+<td><p><br />
+</p></td>
+<td><p><br />
+</p></td>
+<td><p><br />
+</p></td>
+<td><p><br />
+</p></td>
+<td><p>Resets all cooldowns and removes exhausted debuffs on boss start</p></td>
```
Comment (P-Kito) [inline docs/smart_scripts.md:3757]: not on boss start, rather:  [..] when action is fired.

File: docs/smart_scripts.md [hunk 6]
```diff
@@ -3688,6 +3687,75 @@ This is the probability of the event to occur as a percentage from 0-100. So, if
...
+<td><p>SMART_ACTION_VORTEX_SUMMON</p></td>
+<td><p>221</p></td>
+<td><p>entry</p></td>
+<td><p>Duration (ms)<br />
+0 = permanent</p></td>
+<td><p>Spiral scaling</p></td>
+<td><p>Spiral appearance</p></td>
+<td><p>range max</p></td>
+<td><p>phi_delta</p></td>
+<td><p>Vortex = Spiral</p></td>
```
Comment (P-Kito) [inline docs/smart_scripts.md:3740]: > Allows you to summon creature in a customizable spiral(/vortex). Parameters can be confusing, exmaple parameters for testing: 5000 - summon duration, 5 - Spiral Scaling, 25 - spiral appearance, 60 - range max, 40 - phi delta

**Pull request merged**